build/parents/build/pom.xml

```diff
@@ -74,7 +74,7 @@
         <!-- NOTE: Adding new major versions would require to update the compatibility table in `backend-elasticsearch-compatibility` section of `backend-elasticsearch.asciidoc`. -->
         <version.org.opensearch.compatible.expected.text>1.3 or 2.x</version.org.opensearch.compatible.expected.text>
         <!-- The latest version of OpenSearch tested against by default -->
-        <version.org.opensearch.latest>2.9.0</version.org.opensearch.latest>
+        <version.org.opensearch.latest>2.10.0</version.org.opensearch.latest>
```

Comment:
You also need to update `version.org.opensearch.compatible.regularly-tested.text`